File: topico_4/Haskell/4-elemento.hs
-- Recebe uma lista e um numero n e retorna o n-ésimo termo da lista
elemento n xs
    | n < 0 = xs !! negativoAdaptado
    | otherwise = xs !! n
    where
        negativoAdaptado = n + length xs

main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ elemento a b

File: topico_4/Python/4-elemento.py
# Recebe uma lista e um numero n e retorna o n-ésimo termo da lista

def elemento(n, xs):
    for x in xs:
        if(x == xs[n]):
            return x

print(elemento (2, [2,7,3,9]) == 3) #True
print(elemento (0, [2,7,3,9]) == 2) #True
print(elemento (-1, [2,7,3,9]) == 9) #True
print(elemento (-2, [2,7,3,9]) == 3) #True